service-registry-client/src/main/java/no/difi/meldingsutveksling/serviceregistry/security/JWTDecoder.java

```diff
@@ -0,0 +1,60 @@
+package no.difi.meldingsutveksling.serviceregistry.security;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.proc.BadJWSException;
+import no.difi.move.common.oauth.KeystoreHelper;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+
+@Component
+public class JWTDecoder {
+
+    @Qualifier("signingKeystoreHelper")
+    private KeystoreHelper keystoreHelper;
+
+    public String getPayload(String serialized) throws BadJWSException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getPayload" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)